packages/sitecore-jss/src/utils/utils.ts

```diff
@@ -107,10 +107,18 @@ export const enforceCors = (
   res: OutgoingMessage,
   allowedOrigins?: string[]
 ): boolean => {
+  if (!req.headers.origin) {
+    return true;
+  }
+
   const defaultAllowedOrigins = process.env.JSS_ALLOWED_ORIGINS
     ? process.env.JSS_ALLOWED_ORIGINS.replace(' ', '').split(',')
     : [];
   allowedOrigins = defaultAllowedOrigins.concat(allowedOrigins || []);
+  const presetCors = res.getHeader('Access-Control-Allow-Origin');
```

Comment:
Can you, please, explain the case here? I think we are not setting _cors_ header anywhere, or it's set automatically?

Reply:
@ambrauer Is this change also related to the case, when request is made server side? So header is present in a response by default?

Reply:
@ambrauer But in a such case _origin_ will not be present, and it will exit at start of the function

Reply:
@illiakovalenko I'm not quite sure about this use-case, will need to let @art-alexeyenko reply first.

Reply:
@ambrauer @illiakovalenko nextjs may set headers of its own (like we do right now via next config)
but in such case our enforceCors might completely ignore the existing header. it's best to avoid that, and consider the existing CORS value in additon to our list of allowed origins

Reply:
@art-alexeyenko I see, makes sense. Let's add some comments in function, to have better understanding of some checks/actions